lib/zendesk_apps_support/manifest/no_override_hash.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+require 'json'
+
+module ZendeskAppsSupport
+  class Manifest
+    class OverrideError < StandardError
+      class << self
+        def message_for(key, original = nil, attempted = nil)
```

Comment:
Does this actually need to be one method which has multiple multiple modes? splitting it into `message_for_key(key)` and `message_for_pair(key, original, attempted)` would remove most of the complexity in this method, and would make it work properly when `original` or `attempted` is falsy.